### PR TITLE
Add ability to exclude empty repositories in Gitlab

### PR DESCRIPTION
**Pull request by varsanojidan**
[Slack thread](https://sourcegraph.slack.com/archives/C04M5M8GWCX/p1679683552428859)

## Test plan

Added a unit test to cover this scenario

### CODE REVIEW

**Review by cbart — APPROVED** [review]

**Comment by varsanojidan** [comment]
cc: @scjohns

File: internal/repos/gitlab.go [hunk 1]
```diff
@@ -88,9 +89,13 @@ func newGitLabSource(logger log.Logger, svc *types.ExternalService, c *schema.Gi
 	}
 
 	var eb excludeBuilder
+	excludeEmptyRepos := false
```
Comment (mrnugget) [inline internal/repos/gitlab.go:92]: Why not build this into `excludeBuilder`?

**Review by mrnugget — APPROVED** [review]

File: internal/repos/gitlab.go (hunk 1 above)
Comment (varsanojidan) [reply internal/repos/gitlab.go:92]: Like build the ability to exclude by a boolean field in excludeBuilder?

Comment (mrnugget) [reply internal/repos/gitlab.go:92]: (Sorry, missed this while in design sprint call)

Yes, that's what I meant. But - according to our DM in Slack - that sounds like too much effort. Fair enough!

Comment (varsanojidan) [reply internal/repos/gitlab.go:92]: Update: Going to work on this separately from this PR and without reflect :).

**Pull request merged**